tests/testthat/test-mergedatasetsbyvariable.R

```diff
@@ -0,0 +1,216 @@
+context("MergeDataSetsByVariable")
+
+findInstDirFile <- function(file)
+{
+    file.path(system.file("testdata", package = "flipData", mustWork = TRUE),
+              file)
+}
+
+# cola15.sav: Q1_*, Q2, Q3, Q3_3, Attr1 included, with "High self-monitor" cases removed.
+# cola16.sav: Q2, Q3, Q3_3, Q4_A, Q4_B, Q4_C, Attr1 (renamed as PartyID) included,
+#             with "Low self-monitor" cases removed.
+# cola17.sav: Q1_* to Q4_* included, with only "High self-monitor" cases kept.
+# cola18.sav: Q1_* to Q4_* included, with IDs in Attr1 (renamed as Attr1_dup)
+#             modified to include duplicate values.
+
+test_that("Example used for widget test in flipFormat", {
+    expect_error(merge.data.set.by.var.output <- MergeDataSetsByVariable(data.set.names = c(findInstDirFile("cola15.sav"),
+                                                                                            findInstDirFile("cola16.sav")),
+                                                                         id.variables = c("Attr1","PartyID"),
+                                                                         variables.to.include.or.omit = list("Q1_A_c", "Q4_B")), NA)
```

Comment:
Surely there is a better way to indicate that a function should return no error, than by a call to something called "expect error"!  Perhaps `expect_silent()`?  Perhaps write a `function expect_no_error(f) { expect_error(f, NA) }`  (A naive reading of an `expect_error(..., NA)` is that an error is expected, but its exact text/type is unspecified.)

Reply:
Created function expectNoError. The documentation for expect_error says that when NA is passed in, no error is expected.